Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetBrowserFilterModel.cpp

```diff
@@ -86,8 +86,9 @@ namespace AzToolsFramework
         {
             if (m_invalidateFilter)
             {
-                invalidateFilter();
+                beginResetModel();
                 m_invalidateFilter = false;
+                endResetModel();
```

Comment:
Author's note:  See the Qt documentation for a sort filter proxy - they recommend resetting the model when you have large numbers of items in it or when the model changes a lot.

Turns out this is something like 100x faster for us, otherwise we get a flood of "row added" / "row removed" and that causes our other models to reset